File: java/pdfclown.samples.cli/src/org/pdfclown/samples/cli/AdvancedPlainTextExtractionSample.java
package org.pdfclown.samples.cli;

import org.pdfclown.documents.Document;
import org.pdfclown.documents.Page;
import org.pdfclown.files.File;
import org.pdfclown.tools.TextExtractor;

/**
  This sample demonstrates <b>how to retrieve text content</b> from a PDF document.
  If you need further information (such as text style and position), please
  see {@link AdvancedTextExtractionSample}.

  @author Stefano Chizzolini (http://www.stefanochizzolini.it)
  @since 0.0.8
  @version 0.1.0
*/
public class AdvancedPlainTextExtractionSample
  extends Sample
{
  @Override
  public boolean run(
    )
  {
    String filePath = promptPdfFileChoice("Please select a PDF file");

    // 1. Open the PDF file!
    File file;
    try
    {file = new File(filePath);}
    catch(Exception e)
    {throw new RuntimeException(filePath + " file access error.",e);}

    Document document = file.getDocument();

    // 2. Plain text extraction from the document pages.
    TextExtractor extractor = new TextExtractor();
    for(Page page : document.getPages())
    {
      if(!promptNextPage(page, false))
        return false;

      // Extract plain text from the current page!
      System.out.println(extractor.extractPlain(page));
    }

    return true;
  }
}

File: dotNET/pdfclown.samples.cli/src/org/pdfclown/samples/cli/AdvancedPlainTextExtractionSample.cs
using org.pdfclown.documents;
using org.pdfclown.files;
using org.pdfclown.tools;

using System;
using System.Collections.Generic;

namespace org.pdfclown.samples.cli
{
  /**
    <summary>This sample demonstrates how to retrieve text content from a PDF document.</summary>
    <remarks>If you need further information (such as text style and position),
    please see <see cref="AdvancedTextExtractionSample"/>.</remarks>
  */
  public class AdvancedPlainTextExtractionSample
    : Sample
  {
    public override bool Run(
      )
    {
      string filePath = PromptPdfFileChoice("Please select a PDF file");

      // 1. Open the PDF file!
      File file = new File(filePath);
      Document document = file.Document;

      // 2. Plain text extraction from the document pages.
      TextExtractor extractor = new TextExtractor();
      foreach(Page page in document.Pages)
      {
          if(!PromptNextPage(page, false))
          return false;

        // Extract plain text from the current page!
        Console.WriteLine(extractor.ExtractPlain(page));
      }

      return true;
    }
  }
}